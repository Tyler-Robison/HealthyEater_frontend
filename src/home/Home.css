.Home {
    background-image: url('../Images/homepage_img.jpg');
    background-repeat: no-repeat;
    background-size: cover;
    height: 94.1vh;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.Home-div { 
    background-color: white;
    border-radius: 1rem;
    box-shadow: 2px 2px 8px rgba(0, 0, 0, 1);
}

@media (min-width: 1000px) and (max-width: 1200px) {
    .Home-para {
        font-size: 1.2rem !important;
    }
}

@media (min-width: 768px) and (max-width: 1000px) {
    .Home-para {
        font-size: 1rem !important;
    }
}

@media (min-width: 576px) and (max-width: 768px) {
    .Home-para {
        font-size: 0.8rem !important;
    }
}

@media  (max-width: 576px) {
    .Home-para {
        font-size: 0.6rem !important;
    }
}


.Home-h1 {
    margin-top: 0.8rem;
    margin-bottom: 1rem;
}

.Home-para {
    font-size: 1.5rem;
}