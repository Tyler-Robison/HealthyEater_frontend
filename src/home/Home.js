import React, { useContext } from "react";
import GlobalContext from "../context/GlobalContext";
import { Link } from "react-router-dom";
import './Home.css'

/** Homepage - this is where user's are re-directed to upon visiting site or logging in
 * 
 * contains basic information about how the site works
  */
const Home = () => {
    const { currentUser } = useContext(GlobalContext)

    return (
        <div className="Home">
            <div className="row mt-3">
                <div className="Home-div col-12">
                    {currentUser ? <h1>Welcome back {currentUser.username}!</h1> : <h1 className="Home-h1">Welcome to Healthy-Eater!</h1>}
                    {currentUser && currentUser.username === 'guest' && <h5>You are logged into the guest account, this account comes pre-populated with some saved recipes and a mealplan</h5>}
                    {!currentUser && <p className="Home-para"> You aren't logged in, <Link to='/users/login'><b>Login</b></Link> or <Link to='/users/signup'><b>Sign-up</b></Link> to access site</p>}
                    <p className="Home-para"><b>Find Recipes:</b> Search for recipes based on ingredients and nutrients</p>
                    <p className="Home-para"><b>Saved Recipes:</b> View or delete your saved recipes</p>
                    <p className="Home-para"><b>Mealplan:</b> Create your mealplan using saved recipes</p>
                    <p className="Home-para"><b>Calculate Points:</b> Determine your weekly allotment of Weight Watcher Smart Points</p>
                </div>
            </div>
        </div>
    )
}

export default Home;