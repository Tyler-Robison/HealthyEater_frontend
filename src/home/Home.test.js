import { render } from '@testing-library/react';
import Home from './Home';
import { MemoryRouter } from "react-router-dom";
import ContextProvider from '../testContext'

it("renders without crashing", function () {
    <MemoryRouter>
        <Home />
    </MemoryRouter>
});

it("matches snapshot", function () {
    const container = render(
        <MemoryRouter>
            <ContextProvider>
                <Home />
            </ContextProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});