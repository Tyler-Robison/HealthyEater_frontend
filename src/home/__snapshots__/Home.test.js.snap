// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <div
      class="row mt-3"
    >
      <div
        class="Home-div col-12"
      >
        <h1>
          Welcome back Tyler!
        </h1>
        <p
          class="Home-para"
        >
          <b>
            Find Recipes:
          </b>
           Search for recipes based on ingredients and nutrients
        </p>
        <p
          class="Home-para"
        >
          <b>
            Saved Recipes:
          </b>
           View or delete your saved recipes
        </p>
        <p
          class="Home-para"
        >
          <b>
            Mealplan:
          </b>
           Create your mealplan using saved recipes
        </p>
        <p
          class="Home-para"
        >
          <b>
            Calculate Points:
          </b>
           Determine your weekly allotment of Weight Watcher Smart Points
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
